Update kubernetes provider documentation and 8.6 release notes (#2552)
elastic/observability-docs: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-conditions-autodiscover.asciidoc

The condition `${kubernetes.labels.app} == 'redis'` will make the {agent} look for a Pod with the  label `app:redis` within the scope defined in its manifest.

For a list of provider fields that you can use in conditions, refer to <<kubernetes-provider>>.
Some examples of conditions usage are:

> 1. For a pod with label `app.kubernetes.io/name=ingress-nginx` the condition should be `condition: ${kubernetes.labels.app.kubernetes.io/name} == "ingress-nginx"`

Reviewer: Also the numeric list here seems broken in the view https://github.com/elastic/observability-docs/blob/e7771ca322ef8a0a7b1c8cf1cc2493aa62c951d4/docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-conditions-autodiscover.asciidoc

Can you have a look to update it

Author: Fixed

NOTE: All assets (dashboards, ingest pipelines, and so on) related to the Redis integration are not installed. You need to explicitly <<install-uninstall-integration-assets,install them through {kib}>>.

> Conditions can also be used in `Kubernetes` input in order to set the target host dynamically for a targeted Pod based on its labels.

Reviewer 2: ```suggestion
Conditions can also be used in the `Kubernetes` package in order to set the target host dynamically for a targeted Pod based on its labels.
```

NOTE: All assets (dashboards, ingest pipelines, and so on) related to the Redis integration are not installed. You need to explicitly <<install-uninstall-integration-assets,install them through {kib}>>.

> Conditions can also be used in the `Kubernetes` package in order to set the target host dynamically for a targeted Pod based on its labels.

Reviewer 3: by `Kubernetes package` are you referring to the https://docs.elastic.co/en/integrations/kubernetes ?

for standalone we usually use in documentation `input` instead of package, for example - [here](https://observability-docs_2552.docs-preview.app.elstc.co/guide/en/fleet/master/elastic-agent-input-configuration.html) or [here](https://observability-docs_2552.docs-preview.app.elstc.co/guide/en/fleet/master/conditions-based-autodiscover.html#conditions-based-autodiscover)

can we keep it consistent?

Reviewer 3: thank you for clarifications, I don't have a strong opinion on it

Reviewer 2: No strong opinion either :)

Author: Ok as long you don't have a strong opinion I decided to use `input` instead of package to keep consistency because I am not sure if the user at this point is 100% familiar with packages.

  condition: ${kubernetes.labels.component} == 'kube-scheduler'
----

> NOTE: Pods labels and annotations can be used in autodiscover conditions. In the case of labels or annotations that include dots(`.`), they can be used in conditions exactly as

Reviewer 2: ```suggestion
NOTE: Pods' labels and annotations can be used in autodiscover conditions. In the case of labels or annotations that include dots(`.`), they can be used in conditions exactly as
```

> they are defined in the pods. For example `condition: ${kubernetes.labels.app.kubernetes.io/name} == 'ingress-nginx'`. This should not be mistaken with optional dedoted labels and annotations

Reviewer: Do you think can we enhance the examples here?
We like users to understand what is the actual label and what is the condition.

Also an example with annotation?

I think we should so a condition example before and after 8.6.0

Author: To me, there is already an example with the usage of labels in conditions. It is mentioned above that `The condition ${kubernetes.labels.app} == 'redis' will make the Elastic Agent look for a Pod with the label app:redis within the scope defined in its manifest.`
The more complex labels that are not dedoted any more is not a new case. It is just another label to be used. I believe that this note explains it. 
@ChrsMark what do you think ?

Reviewer 2: From my perspective this paragraph achieves the goal to explain how labels should be used in conditions specially regarding the dedoting concpet. Maybe we can one extra example here for an annotation or provide a bunch of examples in a list like:

```
1. To target a Pod labeled with `app: apache` use the condition X
2. To target a Pod with the annotation .... use ....
```
It's up to you to decide how detailed you want to make it.

Author: I can add some examples of conditions with complex labels, annotations and host.ip without having to create detailed examples that do not offer a lot.

Author: @gizas , @ChrsMark see https://github.com/elastic/observability-docs/pull/2552/commits/23a43822c666455bcd10fd70344abfd2ae6a0e46

Reviewer: LGTM now

> they are defined in the pods. For example `condition: ${kubernetes.labels.app.kubernetes.io/name} == 'ingress-nginx'`. This should not be mistaken with optional dedoted labels and annotations

Reviewer: ```This should not be mistaken with optional dedoted labels and annotations
```

This confused me more here. Do you think shall we remove it?

Author: Which line confused you?

Reviewer: See above updated.

I mean do users need this link there.

Author: you mean this link `Elasticsearch(<<kubernetes-provider>>)`. I have added that as it is the page that includes all the info regarding label and annotations dedoting

> they are defined in the pods. For example `condition: ${kubernetes.labels.app.kubernetes.io/name} == 'ingress-nginx'`. This should not be mistaken with optional dedoted labels and annotations

Reviewer 2: ```suggestion
they are defined in the pods. For example `condition: ${kubernetes.labels.app.kubernetes.io/name} == 'ingress-nginx'`. This should not be confused with the dedoted (by default) labels and annotations
```

stored into Elasticsearch(<<kubernetes-provider>>).

WARNING: Before 8.6 release, labels used in autodiscover conditions were dedoted in case `labels.dedot` parameter was set to `true` in Kubernetes Provider 

> configuration. The same did not apply for annotations. This was fixed in 8.6 release. Refer to <<release-notes-8.6.0>>.

Reviewer 2: ```suggestion
configuration (by default `true`). The same did not apply for annotations. This was fixed in 8.6 release. Refer to <<release-notes-8.6.0>>.
```
